<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Several functions are provided for dose-response (or concentration-response) characterization from omics data. DRomics is especially dedicated to omics data obtained using a typical dose-response design, favoring a great number of tested doses (or concentrations) rather than a great number of replicates (no need of replicates). DRomics provides functions 1) to check, normalize and or transform data, 2) to select monotonic or biphasic significantly responding items (e.g. probes, metabolites), 3) to choose the best-fit model among a predefined family of monotonic and biphasic models to describe each selected item, 4) to derive a benchmark dose or concentration and a typology of response from each fitted curve. In the available version data are supposed to be single-channel microarray data in log2, RNAseq data in raw counts, or already pretreated continuous omics data (such as metabolomic data) in log scale. In order to link responses across biological levels based on a common method, DRomics also handles apical data as long as they are continuous and follow a normal distribution for each dose or concentration, with a common standard error. For further details see Delignette-Muller et al (2023) &lt;DOI:10.24072/pcjournal.325&gt; and Larras et al (2018) &lt;DOI:10.1021/acs.est.8b04752&gt;.">
<title>Dose Response for Omics • DRomics</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Dose Response for Omics">
<meta property="og:description" content="Several functions are provided for dose-response (or concentration-response) characterization from omics data. DRomics is especially dedicated to omics data obtained using a typical dose-response design, favoring a great number of tested doses (or concentrations) rather than a great number of replicates (no need of replicates). DRomics provides functions 1) to check, normalize and or transform data, 2) to select monotonic or biphasic significantly responding items (e.g. probes, metabolites), 3) to choose the best-fit model among a predefined family of monotonic and biphasic models to describe each selected item, 4) to derive a benchmark dose or concentration and a typology of response from each fitted curve. In the available version data are supposed to be single-channel microarray data in log2, RNAseq data in raw counts, or already pretreated continuous omics data (such as metabolomic data) in log scale. In order to link responses across biological levels based on a common method, DRomics also handles apical data as long as they are continuous and follow a normal distribution for each dose or concentration, with a common standard error. For further details see Delignette-Muller et al (2023) &lt;DOI:10.24072/pcjournal.325&gt; and Larras et al (2018) &lt;DOI:10.1021/acs.est.8b04752&gt;.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">DRomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.5-2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/DRomics_vignette.html">Overview of the DRomics package</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/aursiber/DRomics/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="dromics-dose-response-for-omics-">DRomics: Dose Response for Omics <img src="reference/figures/hexsticker.png" align="right" style="float:right; height:120px;"><a class="anchor" aria-label="anchor" href="#dromics-dose-response-for-omics-"></a>
</h1></div>
<p><a href="http://cran.r-project.org/package=DRomics" class="external-link"><img src="http://www.r-pkg.org/badges/version-ago/DRomics" alt="CRAN_Release_Badge"></a> <a href="https://cran.r-project.org/package=DRomics" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/DRomics" alt="CRAN Downloads"></a> <a href="https://github.com/aursiber/DRomics/actions" class="external-link"><img src="https://github.com/aursiber/DRomics/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></p>
<p><code>DRomics</code> is a freely available tool for dose-response (or concentration-response) characterization from omics data. It is especially dedicated to omics data obtained using a typical dose-response design, favoring a great number of tested doses (or concentrations) rather than a great number of replicates (no need of replicates to use <code>DRomics</code>).</p>
<p>After a first step which consists in importing, checking and if needed normalizing/transforming the data (step 1), the aim of the proposed workflow is to select monotonic and/or biphasic significantly responsive items (e.g. probes, contigs, metabolites) (step 2), to choose the best-fit model among a predefined family of monotonic and biphasic models to describe the response of each selected item (step 3), and to derive a benchmark dose or concentration from each fitted curve (step 4). Those steps can be performed in R using <code>DRomics</code> functions, or using the shiny application named <code>DRomics-shiny</code>.</p>
<p>In the available version, <code>DRomics</code> supports single-channel microarray data (in log2 scale), RNAseq data (in raw counts) and other continuous omics data such as metabolomics or proteomics (in log scale). In order to link responses across biological levels based on a common method, <code>DRomics</code> also handles continuous apical data as long as they meet the use conditions of least squares regression (homoscedastic Gaussian regression).</p>
<p>As built in the environmental risk assessment context where omics data are more often collected on non-sequenced species or species communities, <code>DRomics</code> does not provide an annotation pipeline. The annotation of items selected by <code>DRomics</code> may be complex in this context, and must be done outside <code>DRomics</code> using databases such as KEGG or Gene Ontology. <code>DRomics</code> functions can then be used to help the interpretation of the workflow results in view of the biological annotation. It enables a multi-omics approach, with the comparison of the responses at the different levels of organization (in view of a common biological annotation). It can also be used to compare the responses at one organization level, but measured under different experimental conditions (e.g. different time points). This interpretation can be performed in R using <code>DRomics</code> functions, or using a second shiny application <code>DRomicsInterpreter-shiny</code>.</p>
<p>All informations about DRomics can also be found at <a href="https://lbbe.univ-lyon1.fr/fr/dromics" target="_blank" class="external-link">https://lbbe.univ-lyon1.fr/fr/dromics</a>.</p>
<p><strong>Keywords</strong> : dose response modelling / benchmark dose (BMD) / environmental risk assessment / transcriptomics / proteomics / metabolomics / toxicogenomics / multi-omics</p>
<div class="section level2">
<h2 id="the-package">The package<a class="anchor" aria-label="anchor" href="#the-package"></a>
</h2>
<p>The <code>limma</code> and <code>DESeq2</code> packages from Bioconductor must be installed for the use of <code>DRomics</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"limma"</span>, <span class="st">"DESeq2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The stable version of <code>DRomics</code> can be installed from CRAN using:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"DRomics"</span><span class="op">)</span></span></code></pre></div>
<p>The development version of <code>DRomics</code> can be installed from GitHub (<code>remotes</code> needed):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span>   </span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"aursiber/DRomics"</span><span class="op">)</span></span></code></pre></div>
<p>Finally load the package in your current R session with the following R command:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lbbe.univ-lyon1.fr/fr/dromics" class="external-link">DRomics</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vignette-and-cheat-sheet">Vignette and cheat sheet<a class="anchor" aria-label="anchor" href="#vignette-and-cheat-sheet"></a>
</h2>
<p>A <strong>vignette</strong> is attached to the <code>DRomics</code> package. This vignette is intended to help users to start using the <code>DRomics</code> package. It is complementary to the reference manual where you can find more details on each function of the package. The first part of this vignette (Main workflow, steps 1 to 4) could also help users of the first shiny application <code>DRomics-shiny</code>. The second part (Help for biological interpretation of <code>DRomics</code> outputs) could also help users of the second shiny application <code>DRomicsInterpreter-shiny</code>.</p>
<p>This vignette can be reached by:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"DRomics_vignette"</span><span class="op">)</span></span></code></pre></div>
<p>Note that, by default, the vignette is not installed when the package is installed through GitHub. The following command (rather long to execute because of the large size of the vignette) will allow you to access the vignette of the development version of the package you installed from GitHub:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"aursiber/DRomics"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>A <strong>cheat sheet</strong> that sum up the DRomics workflow is also available <a href="https://lbbe.univ-lyon1.fr/sites/default/files/media/downloads/dromics.pdf" class="external-link">here</a>.</p>
</div>
<div class="section level2">
<h2 id="two-shiny-apps">Two shiny apps<a class="anchor" aria-label="anchor" href="#two-shiny-apps"></a>
</h2>
<p>The two shiny apps (<code>DRomics-shiny</code> and <code>DRomicsInterpreter-shiny</code>) that work with DRomics are available :</p>
<ul>
<li>on the LBBE shiny server at
<ul>
<li><a href="https://lbbe-shiny.univ-lyon1.fr/DRomics/inst/DRomics-shiny/" class="external-link uri">https://lbbe-shiny.univ-lyon1.fr/DRomics/inst/DRomics-shiny/</a></li>
<li><a href="https://lbbe-shiny.univ-lyon1.fr/DRomics/inst/DRomicsInterpreter-shiny/" class="external-link uri">https://lbbe-shiny.univ-lyon1.fr/DRomics/inst/DRomicsInterpreter-shiny/</a></li>
</ul>
</li>
<li>in the Biosphere cloud, if you or your lab is a partner of the IFB (Institut Français de Bioinformatique), at
<ul>
<li>
<a href="https://biosphere.france-bioinformatique.fr/catalogue/appliance/176/" class="external-link uri">https://biosphere.france-bioinformatique.fr/catalogue/appliance/176/</a> for DRomics-shiny</li>
<li>
<a href="https://biosphere.france-bioinformatique.fr/catalogue/appliance/209/" class="external-link uri">https://biosphere.france-bioinformatique.fr/catalogue/appliance/209/</a> for DRomicsInterpreter-shiny</li>
</ul>
</li>
<li>locally in your R session doing:
<ul>
<li><code>install.packages(c("shiny", "shinyBS", "shinycssloaders", "shinyjs", "shinyWidgets", "sortable"))</code></li>
<li><code>shiny::runApp(system.file("DRomics-shiny", package = "DRomics"))</code></li>
<li><code>shiny::runApp(system.file("DRomicsInterpreter-shiny", package = "DRomics"))</code></li>
</ul>
</li>
</ul>
<p>These shiny apps are runing with the development version of DRomics.</p>
</div>
<div class="section level2">
<h2 id="authors--contacts">Authors &amp; Contacts<a class="anchor" aria-label="anchor" href="#authors--contacts"></a>
</h2>
<p>If you have any need that is not yet covered, any feedback on the package / Shiny app, or any training needs, feel free to email us at <a href="mailto:dromics@univ-lyon1.fr" class="email">dromics@univ-lyon1.fr</a> .</p>
<p>Issues can be reported on <a href="https://github.com/aursiber/DRomics/issues" class="external-link uri">https://github.com/aursiber/DRomics/issues</a> .</p>
<ul>
<li>Elise Billoir: <a href="mailto:elise.billoir@univ-lorraine.fr" class="email">elise.billoir@univ-lorraine.fr</a>
</li>
<li>Marie-Laure Delignette-Muller: <a href="mailto:marielaure.delignettemuller@vetagro-sup.fr" class="email">marielaure.delignettemuller@vetagro-sup.fr</a>
</li>
<li>Floriane Larras: <a href="mailto:floriane.larras@kreatis.eu" class="email">floriane.larras@kreatis.eu</a>
</li>
<li>Mechthild Schmitt-Jansen: <a href="mailto:mechthild.schmitt@ufz.de" class="email">mechthild.schmitt@ufz.de</a>
</li>
<li>Aurélie Siberchicot: <a href="mailto:aurelie.siberchicot@univ-lyon1.fr" class="email">aurelie.siberchicot@univ-lyon1.fr</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use Dromics, you should cite: <br></p>
<p>Delignette-Muller ML, Siberchicot A, Larras F, Billoir E (2023). <em>DRomics, a workflow to exploit dose-response omics data in ecotoxicology</em>. Peer Community Journal. <a href="https://peercommunityjournal.org/articles/10.24072/pcjournal.325/" target="_blank" class="external-link">https://peercommunityjournal.org/articles/10.24072/pcjournal.325/</a></p>
<p>Larras F, Billoir E, Baillard V, Siberchicot A, Scholz S, Wubet T, Tarkka M, Schmitt-Jansen M and Delignette-Muller ML (2018). <em>DRomics : a turnkey tool to support the use of the dose-response framework for omics data in ecological risk assessment</em>. Environmental Science &amp; Technology. <a href="https://pubs.acs.org/doi/10.1021/acs.est.8b04752" target="_blank" class="external-link">https://pubs.acs.org/doi/10.1021/acs.est.8b04752</a> You can find this article at: <a href="https://hal.science/hal-02309919" target="_blank" class="external-link">https://hal.science/hal-02309919</a></p>
<p>You can also look at the following citation for a complete example of use: <br> Larras F, Billoir E, Scholz S, Tarkka M, Wubet T, Delignette-Muller ML, Schmitt-Jansen M (2020). <em>A multi-omics concentration-response framework uncovers novel understanding of triclosan effects in the chlorophyte Scenedesmus vacuolatus</em>. Journal of Hazardous Materials. <a href="https://doi.org/10.1016/j.jhazmat.2020.122727." target="_blank" class="external-link"></a><a href="https://doi.org/10.1016/j.jhazmat.2020.122727" class="external-link uri">https://doi.org/10.1016/j.jhazmat.2020.122727</a>.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=DRomics" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/aursiber/DRomics/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/aursiber/DRomics/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>GPL (&gt;= 2)</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing DRomics</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Marie-Laure Delignette-Muller <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-5453-3994" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Elise Billoir <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-9012-3298" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Aurelie Siberchicot <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7638-8318" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Marie-Laure Delignette-Muller, Elise Billoir, Aurelie Siberchicot.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
