<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Single-channel microarray data in log2 are imported from a .txt file 
(internally imported using the function read.table), 
checked or from a R object of class data.frame 
(see the description 
of argument file for the required format
of data)and normalized (between arrays
normalization).
omicdata is a deprecated version of microarraydata."><title>Import, check and normalization of single-channel microarray data — microarraydata • DRomics</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Import, check and normalization of single-channel microarray data — microarraydata"><meta property="og:description" content="Single-channel microarray data in log2 are imported from a .txt file 
(internally imported using the function read.table), 
checked or from a R object of class data.frame 
(see the description 
of argument file for the required format
of data)and normalized (between arrays
normalization).
omicdata is a deprecated version of microarraydata."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DRomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.5-2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/DRomics_vignette.html">Overview of the DRomics package</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/aursiber/DRomics/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Import, check and normalization of single-channel microarray data</h1>
      
      <div class="d-none name"><code>microarraydata.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Single-channel microarray data in log2 are imported from a .txt file 
(internally imported using the function <code><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></code>), 
checked or from a R object of class <code>data.frame</code> 
(see the description 
of argument <code>file</code> for the required format
of data)and normalized (between arrays
normalization).
<code>omicdata</code> is a deprecated version of <code>microarraydata</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">microarraydata</span><span class="op">(</span><span class="va">file</span>, <span class="va">backgrounddose</span>, check <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  norm.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cyclicloess"</span>, <span class="st">"quantile"</span>, <span class="st">"scale"</span>, <span class="st">"none"</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="fu">omicdata</span><span class="op">(</span><span class="va">file</span>, <span class="va">backgrounddose</span>, check <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  norm.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cyclicloess"</span>, <span class="st">"quantile"</span>, <span class="st">"scale"</span>, <span class="st">"none"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for microarraydata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span><span class="co"># S3 method for microarraydata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">x</span>, range4boxplot <span class="op">=</span> <span class="fl">0</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>file</dt>
<dd><p>The name of the .txt file (e.g. <code>"mydata.txt"</code>)
containing one row per item, with the first column corresponding to the identifier of each item, 
and the other columns giving the responses
of the item for each replicate at each dose or concentration. 
In the first line, after a name for the identifier column, we must have the tested
doses or concentrations in a numeric format for the corresponding replicate 
(for example, if there are triplicates for each treatment, 
the first line could be "item", 0, 0, 0, 0.1, 0.1, 0.1, etc.). 
This file is imported within the function using the function
<code><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></code> with its default field separator (sep argument) and 
its default decimal separator (dec argument at ".").
Alternatively an R object of class <code>data.frame</code> can be directly given in input, 
corresponding to the output
of <code>read.table(file, header = FALSE)</code> on a file described as above.</p></dd>



<dt>backgrounddose</dt>
<dd><p>This argument must be used when there is no dose at zero in the data, to prevent the calculation of the BMD by extrapolation. All doses below  or equal to the value given in backgrounddose will be fixed at 0, so as to be considered at the background level of exposition.</p></dd>



<dt>check</dt>
<dd><p>If TRUE the format of the input file is checked.</p></dd>


<dt>norm.method</dt>
<dd><p>If <code>"none"</code> no normalization is performed, else a 
normalization is performed using the function normalizeBetweenArrays of the 
<code>limma</code> package using the specified method.</p></dd>


<dt>x</dt>
<dd><p>An object of class <code>"microarraydata"</code>.</p></dd>


<dt>range4boxplot</dt>
<dd><p>An argument passed to boxplot(), fixed by default at 0 
  to prevent the producing of very large plot files due to many outliers. 
  Can be put at 1.5 to obtain more classical boxplots.</p></dd>


<dt>...</dt>
<dd><p>further arguments passed to print or plot functions.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function imports the data, checks their format (see the description 
of argument <code>file</code> for the required format
of data) and gives in the <code>print</code> information
that should help the user to check that the coding of data is correct : the tested doses (or concentrations) 
the number of replicates for each dose, the number of items, the identifiers 
of the first 20 items. If the argument <code>norm.method</code> is not <code>"none"</code>,
data are normalized using the function <code><a href="https://rdrr.io/pkg/limma/man/normalizebetweenarrays.html" class="external-link">normalizeBetweenArrays</a></code> of the 
<code>limma</code> package using the specified method : <code>"cyclicloess"</code> (default choice), 
<code>"quantile"</code> or <code>"scale"</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p></p>
<p><code>microarraydata</code> returns an object of class "microarraydata", a list with 9 components:</p>
<dl><dt>data</dt>
<dd><p>the numeric matrix of normalized responses of each item in each replicate
  (one line per item, one column per replicate)</p></dd>


    <dt>dose</dt>
<dd><p>the numeric vector of the tested doses or concentrations corresponding
    to each column of data</p></dd>


    <dt>item</dt>
<dd><p>the character vector of the identifiers of the items, corresponding to
    each line of data</p></dd>


    <dt>design</dt>
<dd><p>a table with the experimental design (tested doses and number of 
    replicates for each dose) for control by the user</p></dd>


    <dt>data.mean</dt>
<dd><p>the numeric matrix of mean responses of each item per dose (mean 
    of the corresponding replicates) (one line per item, one column per unique value 
    of the dose</p></dd>

    
    <dt>data.sd</dt>
<dd><p>the numeric matrix of standard deviations of the response of each 
    item per dose (sd of the corresponding replicates, NA if no replicate) 
    (one line per item, one column per unique value of the dose)</p></dd>


    <dt>norm.method</dt>
<dd><p>The normalization method specified in input</p></dd>


    <dt>data.beforenorm</dt>
<dd><p>the numeric matrix of responses of each item in each replicate
  (one line per item, one column per replicate) before normalization</p></dd>


    <dt>containsNA</dt>
<dd><p>always at FALSE as microarray data are not allowed to contain 
    NA values</p></dd>


</dl><p>The print of a <code>microarraydata</code> object gives the tested doses (or concentrations) 
  and number of replicates for each dose, the number of items, the identifiers 
  of the first 20 items (for check of good coding of data) and the normalization method.
   The plot of a <code>microarraydata</code> object shows the data distribution for each dose or concentration and replicate before and after normalization.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>See <code><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></code> the function used to import data, 
    <code><a href="https://rdrr.io/pkg/limma/man/normalizebetweenarrays.html" class="external-link">normalizeBetweenArrays</a></code> for details about the normalization and 
    <code><a href="RNAseqdata.html">RNAseqdata</a></code>, <code><a href="metabolomicdata.html">continuousomicdata</a></code> and
    <code><a href="continuousanchoringdata.html">continuousanchoringdata</a></code> for other types of data.</p></div>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, and Smyth, GK (2015), <em>limma powers differential expression analyses for RNA-sequencing and microarray studies</em>. Nucleic Acids Research 43, e47.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Marie-Laure Delignette-Muller</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># (1) import, check and normalization of microarray data </span></span></span>
<span class="r-in"><span><span class="co"># (an example on a subsample of a greater data set published in Larras et al. 2018</span></span></span>
<span class="r-in"><span><span class="co"># Transcriptomic effect of triclosan in the chlorophyte Scenedesmus vacuolatus)</span></span></span>
<span class="r-in"><span><span class="va">datafilename</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"transcripto_very_small_sample.txt"</span>, </span></span>
<span class="r-in"><span>  package<span class="op">=</span><span class="st">"DRomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">o</span> <span class="op">&lt;-</span> <span class="fu">microarraydata</span><span class="op">(</span><span class="va">datafilename</span>, check <span class="op">=</span> <span class="cn">TRUE</span>, norm.method <span class="op">=</span> <span class="st">"cyclicloess"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Just wait, the normalization using cyclicloess may take a few minutes.</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Elements of the experimental design in order to check the coding of the data:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Tested doses and number of replicates for each dose:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     0  0.69 1.223 2.148 3.774 6.631 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5     5     5     5     5     5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of items: 100 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Identifiers of the first 20 items:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "1"    "2"    "3"    "4"    "5.1"  "6.1"  "7.1"  "8.1"  "9.1"  "10.1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] "11.1" "12.1" "13.1" "14.1" "15"   "16.1" "17.1" "18.1" "19.1" "20.1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data were normalized between arrays using the following method: cyclicloess </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="microarraydata-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="PCAdataplot.html">PCAdataplot</a></span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="microarraydata-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="PCAdataplot.html">PCAdataplot</a></span><span class="op">(</span><span class="va">o</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="microarraydata-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># If you want to use your own data set just replace datafilename, </span></span></span>
<span class="r-in"><span><span class="co"># the first argument of microarraydata(), </span></span></span>
<span class="r-in"><span><span class="co"># by the name of your data file (e.g. "mydata.txt")</span></span></span>
<span class="r-in"><span><span class="co"># </span></span></span>
<span class="r-in"><span><span class="co"># You should take care that the field separator of this data file is one</span></span></span>
<span class="r-in"><span><span class="co"># of the default field separators recognised by the read.table() function</span></span></span>
<span class="r-in"><span><span class="co"># when it is used with its default field separator (sep argument)</span></span></span>
<span class="r-in"><span><span class="co"># Tabs are recommended.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># (2) normalization with other methods</span></span></span>
<span class="r-in"><span><span class="op">(</span><span class="va">o.2</span> <span class="op">&lt;-</span> <span class="fu">microarraydata</span><span class="op">(</span><span class="va">datafilename</span>, check <span class="op">=</span> <span class="cn">TRUE</span>, norm.method <span class="op">=</span> <span class="st">"quantile"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Elements of the experimental design in order to check the coding of the data:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Tested doses and number of replicates for each dose:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     0  0.69 1.223 2.148 3.774 6.631 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5     5     5     5     5     5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of items: 100 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Identifiers of the first 20 items:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "1"    "2"    "3"    "4"    "5.1"  "6.1"  "7.1"  "8.1"  "9.1"  "10.1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] "11.1" "12.1" "13.1" "14.1" "15"   "16.1" "17.1" "18.1" "19.1" "20.1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data were normalized between arrays using the following method: quantile </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">o.2</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="microarraydata-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="op">(</span><span class="va">o.3</span> <span class="op">&lt;-</span> <span class="fu">microarraydata</span><span class="op">(</span><span class="va">datafilename</span>, check <span class="op">=</span> <span class="cn">TRUE</span>, norm.method <span class="op">=</span> <span class="st">"scale"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Elements of the experimental design in order to check the coding of the data:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Tested doses and number of replicates for each dose:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     0  0.69 1.223 2.148 3.774 6.631 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5     5     5     5     5     5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of items: 100 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Identifiers of the first 20 items:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "1"    "2"    "3"    "4"    "5.1"  "6.1"  "7.1"  "8.1"  "9.1"  "10.1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] "11.1" "12.1" "13.1" "14.1" "15"   "16.1" "17.1" "18.1" "19.1" "20.1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data were normalized between arrays using the following method: scale </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">o.3</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="microarraydata-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Marie-Laure Delignette-Muller, Elise Billoir, Aurelie Siberchicot.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

