<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot of the repartition of trends per group — trendplot • DRomics</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot of the repartition of trends per group — trendplot"><meta property="og:description" content="Provides a plot of the repartition of dose-response trends per group
of items."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DRomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.5-0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/DRomics_vignette.html">Overview of the DRomics package</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/aursiber/DRomics/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot of the repartition of trends per group</h1>
    
    <div class="hidden name"><code>trendplot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Provides a plot of the repartition of dose-response trends per group
of items.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">trendplot</span><span class="op">(</span><span class="va">extendedres</span>, <span class="va">group</span>, <span class="va">facetby</span>, <span class="va">ncol4faceting</span>, add.color <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>extendedres</dt>
<dd><p>the dataframe of results provided by drcfit (fitres)
or bmdcalc (res) 
 or a subset of this data frame (selected lines). This dataframe should be extended 
 with additional columns coming for the group (for example from the functional
 annotation of items) and/or for another level (for example the molecular level),
 and some lines 
 can be replicated if their corresponding item has more than one annotation. 
 This extended dataframe
 must at least contain as results of the dose-response modelling
 the column giving the trend (<code>trend</code>).</p></dd>


<dt>group</dt>
<dd><p>the name of the column of <code>extendedres</code> coding for the groups
on which we want to see the repartition of dose-response trends. 
This column should be a factor ordered as you want the groups to appear
in the plot from bottom up.</p></dd>


<dt>facetby</dt>
<dd><p>optional argument naming the column of <code>extendedres</code> chosen 
  to split the plot in facets using <code><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">ggplot2::facet_wrap</a></code> (no split if omitted).</p></dd>


<dt>ncol4faceting</dt>
<dd><p>number of columns for faceting.</p></dd>


<dt>add.color</dt>
<dd><p>if TRUE a color is added coding for the trend.</p></dd>


</dl></div>
    <div id="value">
    <h2>Value</h2>
     

<p>a ggplot object.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>See <code><a href="bmdplotwithgradient.html">bmdplotwithgradient</a></code> and <code><a href="curvesplot.html">curvesplot</a></code>.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Marie-Laure Delignette-Muller</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># (1)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># An example from the paper published by Larras et al. 2020</span></span></span>
<span class="r-in"><span><span class="co"># in Journal of Hazardous Materials</span></span></span>
<span class="r-in"><span><span class="co"># https://doi.org/10.1016/j.jhazmat.2020.122727</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the dataframe with metabolomic results </span></span></span>
<span class="r-in"><span><span class="va">resfilename</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"triclosanSVmetabres.txt"</span>, package<span class="op">=</span><span class="st">"DRomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">resfilename</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 'data.frame':	31 obs. of  27 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ id              : Factor w/ 31 levels "NAP47_51","NAP_2",..: 2 3 4 5 6 7 8 9 10 11 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ irow            : int  2 21 28 34 38 47 49 51 53 67 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ adjpvalue       : num  6.23e-05 1.11e-05 1.03e-05 1.89e-03 4.16e-03 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ model           : Factor w/ 4 levels "Gauss-probit",..: 2 3 3 2 2 4 2 2 3 3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nbpar           : int  3 2 2 3 3 5 3 3 2 2 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ b               : num  0.4598 -0.0595 -0.0451 0.6011 0.6721 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ c               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ d               : num  5.94 5.39 7.86 6.86 6.21 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ e               : num  -1.648 NA NA -0.321 -0.323 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ f               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ SDres           : num  0.126 0.0793 0.052 0.2338 0.2897 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ typology        : Factor w/ 10 levels "E.dec.concave",..: 2 7 7 2 2 9 2 2 7 7 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ trend           : Factor w/ 4 levels "U","bell","dec",..: 3 3 3 3 3 1 3 3 3 3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ y0              : num  5.94 5.39 7.86 6.86 6.21 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yrange          : num  0.456 0.461 0.35 0.601 0.672 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ maxychange      : num  0.456 0.461 0.35 0.601 0.672 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ xextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD         : num  0.528 1.333 1.154 0.158 0.182 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.zSD         : num  5.82 5.31 7.81 6.62 5.92 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold       : num  NA NA NA NA 0.832 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.xfold       : num  5.35 4.85 7.07 6.17 5.59 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.lower   : num  0.2001 0.8534 0.7519 0.0554 0.081 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.upper   : num  1.11 1.746 1.465 0.68 0.794 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.lower : num  Inf 7.611 Inf 0.561 0.329 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.upper : num  Inf Inf Inf Inf Inf ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nboot.successful: int  957 1000 1000 648 620 872 909 565 1000 1000 ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the dataframe with annotation of each item identified in the previous file</span></span></span>
<span class="r-in"><span><span class="co"># each item may have more than one annotation (-&gt; more than one line)</span></span></span>
<span class="r-in"><span><span class="va">annotfilename</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"triclosanSVmetabannot.txt"</span>, package<span class="op">=</span><span class="st">"DRomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">annotfilename</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 'data.frame':	84 obs. of  2 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ metab.code: Factor w/ 31 levels "NAP47_51","NAP_2",..: 2 3 4 4 4 4 5 6 7 8 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ path_class: Factor w/ 9 levels "Amino acid metabolism",..: 5 3 3 2 6 8 5 5 5 5 ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Merging of both previous dataframes</span></span></span>
<span class="r-in"><span><span class="co"># in order to obtain an extenderes dataframe</span></span></span>
<span class="r-in"><span><span class="va">extendedres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">res</span>, y <span class="op">=</span> <span class="va">annot</span>, by.x <span class="op">=</span> <span class="st">"id"</span>, by.y <span class="op">=</span> <span class="st">"metab.code"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">extendedres</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         id irow    adjpvalue       model nbpar           b  c        d</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 NAP47_51   46 7.158246e-04      linear     2 -0.05600559 NA 7.343571</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    NAP_2    2 6.232579e-05 exponential     3  0.45981242 NA 5.941896</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   NAP_23   21 1.106958e-05      linear     2 -0.05946618 NA 5.387252</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   NAP_30   28 1.028343e-05      linear     2 -0.04507832 NA 7.859109</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   NAP_30   28 1.028343e-05      linear     2 -0.04507832 NA 7.859109</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   NAP_30   28 1.028343e-05      linear     2 -0.04507832 NA 7.859109</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           e  f      SDres     typology trend       y0    yrange maxychange</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        NA NA 0.12454183        L.dec   dec 7.343571 0.4346034  0.4346034</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 -1.647958 NA 0.12604568 E.dec.convex   dec 5.941896 0.4556672  0.4556672</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3        NA NA 0.07929266        L.dec   dec 5.387252 0.4614576  0.4614576</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4        NA NA 0.05203245        L.dec   dec 7.859109 0.3498078  0.3498078</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5        NA NA 0.05203245        L.dec   dec 7.859109 0.3498078  0.3498078</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6        NA NA 0.05203245        L.dec   dec 7.859109 0.3498078  0.3498078</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   xextrem yextrem   BMD.zSD  BMR.zSD BMD.xfold BMR.xfold BMD.zSD.lower</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      NA      NA 2.2237393 7.219029        NA  6.609214     0.9785095</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      NA      NA 0.5279668 5.815850        NA  5.347706     0.2000881</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      NA      NA 1.3334076 5.307960        NA  4.848527     0.8533711</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      NA      NA 1.1542677 7.807077        NA  7.073198     0.7518588</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      NA      NA 1.1542677 7.807077        NA  7.073198     0.7518588</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      NA      NA 1.1542677 7.807077        NA  7.073198     0.7518588</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   BMD.zSD.upper BMD.xfold.lower BMD.xfold.upper nboot.successful</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1      4.068699             Inf             Inf             1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2      1.109559             Inf             Inf              957</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3      1.746010        7.610936             Inf             1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4      1.464998             Inf             Inf             1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5      1.464998             Inf             Inf             1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6      1.464998             Inf             Inf             1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                    path_class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                            Lipid metabolism</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                            Lipid metabolism</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                     Carbohydrate metabolism</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4                     Carbohydrate metabolism</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 Biosynthesis of other secondary metabolites</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6                          Membrane transport</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># (1.a) Trendplot by pathway</span></span></span>
<span class="r-in"><span><span class="fu">trendplot</span><span class="op">(</span><span class="va">extendedres</span>, group <span class="op">=</span> <span class="st">"path_class"</span><span class="op">)</span> </span></span>
<span class="r-plt img"><img src="trendplot-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># (1.b) Trendplot by pathway without color</span></span></span>
<span class="r-in"><span><span class="fu">trendplot</span><span class="op">(</span><span class="va">extendedres</span>, group <span class="op">=</span> <span class="st">"path_class"</span>, add.color <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></span>
<span class="r-plt img"><img src="trendplot-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># (1.c) Reordering of the groups before plotting</span></span></span>
<span class="r-in"><span><span class="va">extendedres</span><span class="op">$</span><span class="va">path_class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">extendedres</span><span class="op">$</span><span class="va">path_class</span>, </span></span>
<span class="r-in"><span>                levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">extendedres</span><span class="op">$</span><span class="va">path_class</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">trendplot</span><span class="op">(</span><span class="va">extendedres</span>, group <span class="op">=</span> <span class="st">"path_class"</span>, add.color <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></span>
<span class="r-plt img"><img src="trendplot-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># (2) </span></span></span>
<span class="r-in"><span><span class="co"># An example with two molecular levels</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co">### Rename metabolomic results</span></span></span>
<span class="r-in"><span><span class="va">metabextendedres</span> <span class="op">&lt;-</span> <span class="va">extendedres</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Import the dataframe with transcriptomic results </span></span></span>
<span class="r-in"><span><span class="va">contigresfilename</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"triclosanSVcontigres.txt"</span>, package <span class="op">=</span> <span class="st">"DRomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">contigres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">contigresfilename</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">contigres</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 'data.frame':	447 obs. of  27 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ id              : Factor w/ 447 levels "c00134","c00276",..: 1 2 3 4 5 6 7 8 9 10 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ irow            : int  2802 39331 41217 52577 52590 53968 54508 57776 58705 60306 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ adjpvalue       : num  2.76e-04 9.40e-07 2.89e-06 1.88e-03 1.83e-03 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ model           : Factor w/ 4 levels "Gauss-probit",..: 3 2 2 2 2 2 3 2 1 3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nbpar           : int  2 3 3 3 3 3 2 3 4 2 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ b               : num  -0.21794 1.49944 1.40817 0.00181 1.48605 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ c               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ d               : num  10.9 12.4 12.4 16.4 15.3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ e               : num  NA -2.2 -2.41 1.15 -2.31 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ f               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ SDres           : num  0.417 0.287 0.281 0.145 0.523 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ typology        : Factor w/ 10 levels "E.dec.concave",..: 7 2 2 4 2 2 7 1 5 8 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ trend           : Factor w/ 4 levels "U","bell","dec",..: 3 3 3 4 3 3 3 3 1 4 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ y0              : num  10.9 12.4 12.4 16.4 15.3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yrange          : num  1.445 1.426 1.319 0.567 1.402 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ maxychange      : num  1.445 1.426 1.319 0.567 1.402 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ xextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD         : num  1.913 0.467 0.536 5.073 1.004 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.zSD         : num  10.4 12.1 12.1 16.6 14.8 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold       : num  4.98 3.88 5.13 NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.xfold       : num  9.77 11.19 11.17 18.05 13.8 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.lower   : num  1.255 0.243 0.282 2.65 0.388 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.upper   : num  2.759 0.825 0.925 5.573 2.355 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.lower : num  3.94 2.32 2.79 Inf 3.06 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.upper : num  Inf Inf Inf Inf Inf ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nboot.successful: int  500 497 495 332 466 469 500 321 260 500 ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Import the dataframe with functional annotation (or any other descriptor/category </span></span></span>
<span class="r-in"><span><span class="co"># you want to use, here KEGG pathway classes) </span></span></span>
<span class="r-in"><span><span class="va">contigannotfilename</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"triclosanSVcontigannot.txt"</span>, package <span class="op">=</span> <span class="st">"DRomics"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">contigannot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">contigannotfilename</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">contigannot</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 'data.frame':	562 obs. of  2 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ contig    : Factor w/ 447 levels "c00134","c00276",..: 1 2 3 4 5 6 7 8 9 10 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ path_class: Factor w/ 17 levels "Amino acid metabolism",..: 3 11 11 15 8 4 3 4 8 2 ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Merging of both previous dataframes   </span></span></span>
<span class="r-in"><span><span class="va">contigextendedres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">contigres</span>, y <span class="op">=</span> <span class="va">contigannot</span>, by.x <span class="op">=</span> <span class="st">"id"</span>, by.y <span class="op">=</span> <span class="st">"contig"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># to see the structure of this dataframe</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">contigextendedres</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 'data.frame':	562 obs. of  28 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ id              : Factor w/ 447 levels "c00134","c00276",..: 1 2 3 4 5 6 7 8 9 10 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ irow            : int  2802 39331 41217 52577 52590 53968 54508 57776 58705 60306 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ adjpvalue       : num  2.76e-04 9.40e-07 2.89e-06 1.88e-03 1.83e-03 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ model           : Factor w/ 4 levels "Gauss-probit",..: 3 2 2 2 2 2 3 2 1 3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nbpar           : int  2 3 3 3 3 3 2 3 4 2 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ b               : num  -0.21794 1.49944 1.40817 0.00181 1.48605 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ c               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ d               : num  10.9 12.4 12.4 16.4 15.3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ e               : num  NA -2.2 -2.41 1.15 -2.31 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ f               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ SDres           : num  0.417 0.287 0.281 0.145 0.523 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ typology        : Factor w/ 10 levels "E.dec.concave",..: 7 2 2 4 2 2 7 1 5 8 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ trend           : Factor w/ 4 levels "U","bell","dec",..: 3 3 3 4 3 3 3 3 1 4 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ y0              : num  10.9 12.4 12.4 16.4 15.3 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yrange          : num  1.445 1.426 1.319 0.567 1.402 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ maxychange      : num  1.445 1.426 1.319 0.567 1.402 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ xextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD         : num  1.913 0.467 0.536 5.073 1.004 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.zSD         : num  10.4 12.1 12.1 16.6 14.8 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold       : num  4.98 3.88 5.13 NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.xfold       : num  9.77 11.19 11.17 18.05 13.8 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.lower   : num  1.255 0.243 0.282 2.65 0.388 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.upper   : num  2.759 0.825 0.925 5.573 2.355 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.lower : num  3.94 2.32 2.79 Inf 3.06 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.upper : num  Inf Inf Inf Inf Inf ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nboot.successful: int  500 497 495 332 466 469 500 321 260 500 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ path_class      : Factor w/ 17 levels "Amino acid metabolism",..: 3 11 11 15 8 4 3 4 8 2 ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Merge metabolomic and transcriptomic results</span></span></span>
<span class="r-in"><span><span class="va">extendedres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">metabextendedres</span>, <span class="va">contigextendedres</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">extendedres</span><span class="op">$</span><span class="va">molecular.level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"metabolites"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">metabextendedres</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                              <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"contigs"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">contigextendedres</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">extendedres</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 'data.frame':	646 obs. of  29 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ id              : Factor w/ 478 levels "NAP47_51","NAP_2",..: 1 2 3 4 4 4 4 5 6 7 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ irow            : int  46 2 21 28 28 28 28 34 38 47 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ adjpvalue       : num  7.16e-04 6.23e-05 1.11e-05 1.03e-05 1.03e-05 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ model           : Factor w/ 4 levels "Gauss-probit",..: 3 2 3 3 3 3 3 2 2 4 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nbpar           : int  2 3 2 2 2 2 2 3 3 5 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ b               : num  -0.056 0.4598 -0.0595 -0.0451 -0.0451 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ c               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ d               : num  7.34 5.94 5.39 7.86 7.86 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ e               : num  NA -1.65 NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ f               : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ SDres           : num  0.1245 0.126 0.0793 0.052 0.052 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ typology        : Factor w/ 10 levels "E.dec.concave",..: 7 2 7 7 7 7 7 2 2 9 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ trend           : Factor w/ 4 levels "U","bell","dec",..: 3 3 3 3 3 3 3 3 3 1 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ y0              : num  7.34 5.94 5.39 7.86 7.86 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yrange          : num  0.435 0.456 0.461 0.35 0.35 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ maxychange      : num  0.435 0.456 0.461 0.35 0.35 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ xextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ yextrem         : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD         : num  2.224 0.528 1.333 1.154 1.154 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.zSD         : num  7.22 5.82 5.31 7.81 7.81 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold       : num  NA NA NA NA NA ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMR.xfold       : num  6.61 5.35 4.85 7.07 7.07 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.lower   : num  0.979 0.2 0.853 0.752 0.752 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.zSD.upper   : num  4.07 1.11 1.75 1.46 1.46 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.lower : num  Inf Inf 7.61 Inf Inf ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ BMD.xfold.upper : num  Inf Inf Inf Inf Inf ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ nboot.successful: int  1000 957 1000 1000 1000 1000 1000 648 620 872 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ path_class      : Factor w/ 18 levels "Translation",..: 5 5 7 7 8 4 2 5 5 5 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ molecular.level : Factor w/ 2 levels "contigs","metabolites": 2 2 2 2 2 2 2 2 2 2 ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### trend plot of both molecular levels</span></span></span>
<span class="r-in"><span><span class="co"># optional inverse alphabetic ordering of groups for the plot</span></span></span>
<span class="r-in"><span><span class="va">extendedres</span><span class="op">$</span><span class="va">path_class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">extendedres</span><span class="op">$</span><span class="va">path_class</span>, </span></span>
<span class="r-in"><span>                levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">extendedres</span><span class="op">$</span><span class="va">path_class</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">trendplot</span><span class="op">(</span><span class="va">extendedres</span>, group <span class="op">=</span> <span class="st">"path_class"</span>, facetby <span class="op">=</span> <span class="st">"molecular.level"</span><span class="op">)</span> </span></span>
<span class="r-plt img"><img src="trendplot-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Marie-Laure Delignette-Muller, Elise Billoir, Aurelie Siberchicot.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

